bookkeeper-server/src/test/java/org/apache/bookkeeper/proto/BookieProtocolTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.proto;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.junit.Assert.assertEquals;
+
+import com.google.protobuf.ByteString;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.util.ReferenceCountUtil;
+import org.apache.bookkeeper.util.ByteBufList;
+import org.junit.Test;
+
+public class BookieProtocolTest {
+
+    @Test
+    public void testAddRequestTouch() {
+        byte[] data = new byte[4];
+        ByteBuf byteBuf = Unpooled.wrappedBuffer(data);
+
+        ByteBufList byteBufList = ByteBufList.get(byteBuf.retainedSlice());
+
+        BookieProtocol.AddRequest addRequest = BookieProtocol.AddRequest.create(BookieProtocol.CURRENT_PROTOCOL_VERSION,
+                0, 0, (short) 1, "test".getBytes(UTF_8), byteBufList);
+        assertEquals(ReferenceCountUtil.touch(addRequest).hashCode(), addRequest.hashCode());
```

Comment:
we should verify that the underlying `ByteBuf.retain/touch` is called, not the returned object. It can be easily addressed using `Mockito.verify()`

ReferenceCountUtil methods don't change the object you pass so these assertions aren't useful